internal/models/user.go

```diff
@@ -531,17 +531,17 @@ func FindUsersInAudience(tx *storage.Connection, aud string, pageParams *Paginat
 func FindUserByEmailChangeCurrentAndAudience(tx *storage.Connection, email, token, aud string) (*User, error) {
 	return findUser(
 		tx,
-		"instance_id = ? and LOWER(email) = ? and email_change_token_current = ? and aud = ? and is_sso_user = false",
-		uuid.Nil, strings.ToLower(email), token, aud,
+		"instance_id = ? and LOWER(email) = ? and aud = ? and is_sso_user = false and (email_change_token_current like 'pkce_%' || ? or email_change_token_current = ?)",
```

Comment:
Are you sure you mean `like 'pkce_%'`? Feels like it's not supposed to be there, because it will match any token starting with pkce and not the one you're looking for?

Reply:
it wouldn't match any token because it's concatenated with the `token` - but yeah i don't think the wildcard is necessary so i've removed it